### PR TITLE
Support grayscale image in resize

### PR DESCRIPTION
**Pull request by Hakuyume**
`resize` function  with the `Pillow` backend can not treat grayscale image.
I add a test and fix this bug.

### CODE REVIEW

**Comment by Hakuyume** [comment]
I found `cv2` backend also can not  treat grayscale image.
I fixed this bug.

File: chainercv/transforms/image/resize.py [hunk 1]
```diff
@@ -1,13 +1,17 @@
+import numpy
 import warnings
 
 try:
     import cv2
 
     def _resize(img, size):
-        return cv2.resize(img, dsize=size)
+        img = cv2.resize(img, dsize=size)
+        if len(img.shape) == 3:
```
Comment (yuyu2172) [inline chainercv/transforms/image/resize.py]: Could you add an argument that explains this condition?
I thought it was not intuitive that `cv2` changed shape of img from `H, W, 1` to `H, W`.

So, add something like
"# When channel size of the image is one, the resized image returned by cv2 is two dimensional array.

**Comment by yuyu2172** [comment]
Thanks. Nice PR.

**Comment by yuyu2172** [comment]
LGTM

**Pull request merged**